File: app/src/main/java/com/my/televip/virtuals/SQLite/SQLiteDatabase.java
package com.my.televip.virtuals.SQLite;

import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;

import de.robv.android.xposed.XposedHelpers;

public class SQLiteDatabase {

    Object sqLiteDatabase;

    public SQLiteDatabase(Object obj){ sqLiteDatabase = obj; }

    public SQLiteCursor queryFinalized(String s, Object[] objects){
        return new SQLiteCursor(XposedHelpers.callMethod(sqLiteDatabase, Obfuscate.getMethodName("SQLiteDatabase", "queryFinalized"), s, objects));
    }

    public SQLitePreparedStatement executeFast(String s){
        return new SQLitePreparedStatement(XposedHelpers.callMethod(sqLiteDatabase, Obfuscate.getMethodName("SQLiteDatabase", "executeFast"), s));
    }

}

File: app/src/main/java/com/my/televip/virtuals/SQLite/SQLiteDatabase.kt
package com.my.televip.virtuals.SQLite

import com.my.televip.obfuscate.Obfuscate
import de.robv.android.xposed.XposedHelpers

class SQLiteDatabase(@JvmField val sqLiteDatabase: Any?) {

    fun queryFinalized(s: String?, objects: Array<Any?>): SQLiteCursor {
        val target = sqLiteDatabase ?: return SQLiteCursor(null)
        val result = XposedHelpers.callMethod(target, Obfuscate.getMethodName("SQLiteDatabase", "queryFinalized"), s, objects)
        return SQLiteCursor(result)
    }

    fun executeFast(s: String?): SQLitePreparedStatement {
        val target = sqLiteDatabase ?: return SQLitePreparedStatement(null)
        val result = XposedHelpers.callMethod(target, Obfuscate.getMethodName("SQLiteDatabase", "executeFast"), s)
        return SQLitePreparedStatement(result)
    }
}
